### ISSUE TITLE
feature request: Allow lazy loading

### ISSUE THREAD

**Issue by januswel** [ArnaudBarre/eslint-plugin-react-refresh#40]
Now, the rule raises errors with the following codes

```typescript
const LazyComponent = React.lazy(
  () => import("./MyComponent"),
);
```

```
Fast refresh only works when a file only exports components. Move your component(s) to a separate file.eslint(react-refresh/only-export-components)
```

Can we modify to determine this as the correct code?

**Comment by ArnaudBarre** [comment]
Can you give more context? This code creates a component so it should follow some Fast refresh rules

**Comment by januswel** [comment]
I made a repo to repro

https://github.com/januswel/react-refresh-sample

<img width="1062" alt="image" src="https://github.com/ArnaudBarre/eslint-plugin-react-refresh/assets/43239/44025273-9d1a-49c7-9786-2763fde64d79">

This problem may occur with `RouteObject`s react-router v6

**Comment by ArnaudBarre** [comment]
This is the same issue as https://github.com/ArnaudBarre/eslint-plugin-react-refresh/issues/25

I need to dig deeper to see how it really behaves at runtime. Given that SWC and Babel generate some fast refresh code for that for now it's safer to consider that `React.lazy` calls are like creating a wrapper component (which is the case)

**Comment by januswel** [comment]
@ArnaudBarre 
Thank you for your comment
I saw #25 and consider to use workarounds from you

**Comment by ArnaudBarre** [comment]
I'm closing this one for the same reason as #25, see https://github.com/ArnaudBarre/eslint-plugin-react-refresh/issues/25#issuecomment-2496199809
